### PR TITLE
Add copy to QuantumWorld

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds a way to copy the state of a QuantumWorld.

### CODE REVIEW

File: unitary/alpha/quantum_world_test.py [hunk 1]
```diff
@@ -284,6 +284,50 @@ def test_undo(simulator, compile_to_qubits):
     assert all(result[0] == Light.GREEN for result in results)
 
 
+def test_copy():
```
Comment (smitsanghavi) [inline unitary/alpha/quantum_world_test.py:287]: Can you also add the `compile_to_qubits` parameterization to ensure that copying does not break that somehow?

**Review by smitsanghavi — CHANGES_REQUESTED** [review]

File: unitary/alpha/quantum_world_test.py (hunk 1 above)
Comment (dstrain115) [reply unitary/alpha/quantum_world_test.py]: Done.

Comment (smitsanghavi) [reply unitary/alpha/quantum_world_test.py]: Still need to initialize the `QuantumWorld` with the param.

Comment (dstrain115) [reply unitary/alpha/quantum_world_test.py]: Oh yeah, of course I do.  Thanks for all the reviews!

**Review by smitsanghavi — APPROVED** [review]

File: unitary/alpha/quantum_world_test.py (hunk 1 above)
Comment (smitsanghavi) [reply unitary/alpha/quantum_world_test.py]: No problemo!

**Pull request merged**